modules/rostests/apitests/user32/CharFuncs.c

```diff
@@ -0,0 +1,686 @@
+/*
+ * PROJECT:     ReactOS API tests
+ * LICENSE:     GPL-2.0-or-later (https://spdx.org/licenses/GPL-2.0-or-later)
+ * PURPOSE:     Tests for Char* functions
+ * COPYRIGHT:   Copyright 2022 Stanislav Motylkov <x86corez@gmail.com>
+ */
+
+#include "precomp.h"
+
+#include <winnls.h>
+#include <ndk/rtlfuncs.h>
+#include <pseh/pseh2.h>
+#include <strsafe.h>
+#include <versionhelpers.h>
+
+#define INVALID_PTR_OFF(x)  ((PVOID)(ULONG_PTR)(0xdeadbeefdeadbeefULL + x))
+#define INVALID_PTR         INVALID_PTR_OFF(0)
+#define TEST_ACP            1252
+
+typedef enum
+{
+    testLen,
+    testOffs,
+    testBoth,
+} TEST_TYPE;
+
+/* Dynamic allocation tests */
+typedef struct
+{
+    TEST_TYPE testType;
+    LPWSTR lpszStart;
+    LPWSTR lpszCurrent;
+    INT iOffset;
+    INT iResOffset;
+    INT iResOffsetNeg;
+    BOOL relStart;
+    BOOL bWideOnly;
+} TESTS_CHARPREV;
+
+TESTS_CHARPREV TestCharPrev[] =
+{
+    {testLen, L"C:\\ReactOS", NULL, 0, 9, 9, TRUE, FALSE},
+    {testOffs, L"abcdefghijk", NULL, 11, 10, 10, TRUE, FALSE},
+    {testOffs, L"test a´^~¯", NULL, 10, 9, 9, TRUE, FALSE},
+    {testOffs, L"test å", NULL, 6, 5, 5, TRUE, FALSE},
+    {testBoth, L"C:\\ReactOS", L"", 0, -1, 0, FALSE, FALSE},
+    {testBoth, L"C:\\ReactOS\\", L"C:\\ReactOS", 0, -1, 0, FALSE, FALSE},
+    {testBoth, L"C:\\ReactOS\\", L"ReactOS", 0, -1, 0, FALSE, FALSE},
+};
+
+TESTS_CHARPREV TestCharPrev_XP[] =
+{
+    /* XP/2003 treat diacritics as normal characters */
+    {testOffs, L"test a\x030a", NULL, 7, 6, 6, TRUE, TRUE},
+    {testOffs, L"test a\x0301\x0302\x0303\x0304", NULL, 10, 9, 9, TRUE, TRUE},
+    {testOffs, L"test a\x0301\x0302\x0303\x0304", NULL, 9, 8, 8, TRUE, TRUE},
+    //{testOffs, L"test a\x0301\x0302\x0303\x0304", NULL, 8, 7, 7, TRUE, TRUE}, // exception on ReactOS
```

Comment:
This can be uncommented now, as there was a bug in the test, which is now fixed.

I will uncomment it before merging.